src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -649,7 +647,7 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 	 */
 	if (ContainsUnionSubquery(originalQuery))
 	{
-		if (!SafeToPushdownUnionSubquery(relationRestrictionContext))
+		if (!SafeToPushdownUnionSubquery(plannerRestrictionContext))
```

Comment:
The error message could be very confusing now :/ Would you mind changing it so that it also mentions about the partition key joins?